gax-grpc/src/main/java/com/google/api/gax/grpc/ChannelPool.java

```diff
@@ -44,17 +46,81 @@
  * <p>Package-private for internal use.
  */
 class ChannelPool extends ManagedChannel {
-  private final ImmutableList<ManagedChannel> channels;
+  // refresh every 55 minutes
+  private static long refreshPeriod = 55 * 60;
+  // spread out the refresh between channels
+  private static long refreshDelay;
+  // maximum seconds to wait for old channel to terminate
+  private static long terminationWait = 60;
+
+  private final List<ManagedChannel> channels;
```

Comment:
Does this need to volatile? I know you aren't modifying the underlying array, but there are different threads that are changing elements. Something needs to flush the cache line